examples/sst2/train.py

```diff
@@ -0,0 +1,272 @@
+      labels = jnp.expand_dims(labels, 1)
+    loss = jnp.mean(
+        sigmoid_cross_entropy_with_logits(labels=labels, logits=logits))
+    return loss, (logits, new_model_state)
+
+  grad_fn = jax.value_and_grad(loss_fn, has_aux=True)
+  value, grad = grad_fn(optimizer.target)
+  (_, (logits, new_model_state)) = value
+  optimizer = optimizer.apply_gradient(grad)
+
+  metrics = compute_metrics(labels=batch['label'], logits=logits)
+  return optimizer, metrics, new_model_state
+
+
+@functools.partial(jax.jit, static_argnums=(0,))
+def eval_step(model: nn.Module, params: Dict[str, Any],
+              model_state: Mapping[str, Any], batch: Dict[str, Array],
+              rngs: Dict[str, Any]) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+  """Evaluate for a single step. Model should be in deterministic mode."""
+  variables = {'params': params, **model_state}
+  logits, new_model_state = model.apply(
+      variables, batch['token_ids'], batch['length'],
+      rngs=rngs,
+      mutable=list(model_state.keys()))
+  metrics = compute_metrics(labels=batch['label'], logits=logits)
+  return metrics, new_model_state
+
+
+def normalize_batch_metrics(
+        batch_metrics: Sequence[Dict[str, Any]]) -> Dict[str, Any]:
+  """Consolidates and normalizes a list of per-batch metrics dicts."""
+  # Here we sum the metrics that were already summed per batch.
+  metric_names = batch_metrics[0].keys()
+  summed_metrics = {
+      k: np.sum([metrics[k] for metrics in batch_metrics]) for k in metric_names
+  }
+  # Divide each metric by the total number of items in the data set.
+  total = np.float(summed_metrics.pop('count'))
+  metrics = jax.tree_map(lambda x: x.item() / total, summed_metrics)
+  return metrics
+
+
+def evaluate_model(
+        model: nn.Module,
+        params: Dict[str, Any],
+        model_state: Mapping[str, Any],
+        batches: Union[Iterable[Example], tf.data.Dataset],
+        epoch: int,
+        rngs: Optional[Dict[str, Any]] = None) -> Tuple[Dict[str, Any], Mapping[str, Any]]:
+  """Evaluate a model on a dataset."""
+  batch_metrics = []
+  for i, batch in enumerate(batches):
+    # pylint: disable=protected-access
+    batch = jax.tree_map(lambda x: x._numpy(), batch)
+    if rngs is not None:  # New RNG for each step.
+      rngs = {name: jax.random.fold_in(rng, i) for name, rng in rngs.items()}
+
+    metrics, model_state = eval_step(model, params, model_state, batch, rngs)
+    batch_metrics.append(metrics)
+
+  batch_metrics = jax.device_get(batch_metrics)
+  metrics = normalize_batch_metrics(batch_metrics)
+  logging.info('eval  epoch %03d loss %.4f accuracy %.2f', epoch,
+               metrics['loss'], metrics['accuracy'] * 100)
+  return metrics, model_state
+
+
+def train_epoch(model, optimizer, model_state, train_batches, epoch, rngs):
+  """Train for a single epoch."""
+  batch_metrics = []
+  for batch in train_batches:
+    # pylint: disable=protected-access
+    batch = jax.tree_map(lambda x: x._numpy(), batch)
```

Comment:
I'm not 100% sure but I think adding it like this will disable it for the rest of the file. I think you should either explicitly re-enable it after the call, or do this:

```suggestion
    batch = jax.tree_map(lambda x: x._numpy(), batch)  # pylint: disable=protected-access
```